specification/sections/05-events.md

```diff
@@ -28,9 +28,17 @@ graph
 
 > The `provider` **MAY** define a mechanism for signaling the occurrence of one of a set of events, including `PROVIDER_READY`, `PROVIDER_ERROR`, `PROVIDER_CONFIGURATION_CHANGED` and `PROVIDER_STALE`, with a `provider event details` payload. 
 
+Providers cannot emit `PROVIDER_CONTEXT_CHANGED` or `RECONCILING` event.
+These are emitted only by the SDK during context reconciliation.
+
 If available, native event-emitter or observable/observer language constructs can be used.
 
-see: [provider event types](../types.md#provider-events), [`event details`](../types.md#provider-event-details).
+When a provider is unable to evaluate flags (perhaps due to loss of connection with a remote service) the provider can signal this by emitting a `PROVIDER_ERROR` event.
+If the error state is irrecoverable, the `PROVIDER_FATAL` error code can be used.
+When it recovers, it can emit a `PROVIDER_READY` event.
```

Comment:
```suggestion
When it recovers, it can emit a `PROVIDER_READY` event.
If the error state is irrecoverable, the `PROVIDER_FATAL` error code can be used.
```

I would swap these lines because a `PROVIDER_FATAL` error code isn't recoverable but a `PROVIDER_ERROR` is.